config/base/derive/src/configurable.rs

```diff
@@ -0,0 +1,228 @@
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{Attribute, Ident, Type, TypePath};
+
+use super::utils::{
+    extract_field_attrs, extract_field_idents, extract_field_types, field_has_as_str_attr,
+    field_has_inner_attr, field_has_option_type, gen_field_env, gen_lvalues, get_env_prefix,
+    get_inner_type, StructWithFields,
+};
+
+pub(super) fn impl_configurable(ast: &StructWithFields) -> TokenStream {
+    let StructWithFields {
+        ident: parent_name,
+        vis,
+        generics,
+        ..
+    } = ast;
+    let proxy_name = format_ident!("{}Proxy", parent_name);
+
+    let prefix = get_env_prefix(ast);
+
+    let field_idents = extract_field_idents(&ast.fields);
+
+    let field_attrs = extract_field_attrs(&ast.fields);
+
+    let field_ty = extract_field_types(&ast.fields);
+    let inner = field_has_inner_attr(&field_attrs);
+
+    let field_env = gen_field_env(&field_idents, &prefix);
+
+    let as_str = field_has_as_str_attr(&field_attrs);
+    let option = field_has_option_type(&field_ty);
+
+    let (_lvalue_read, lvalue_write) = gen_lvalues(&field_ty, &field_idents);
+
+    // Doing for parent struct right now instead of the proxy
+    // TODO: when generating for proxy, replace types and option with proxy's values
+    let load_env_impl = impl_load_env(
+        parent_name,
+        &field_idents,
+        &inner,
+        &option,
+        &lvalue_write,
+        &as_str,
+        &field_ty,
+        &field_env,
+    );
+
+    let disk_impl = impl_load_from_disk(&proxy_name);
+    let proxy_fields = gen_proxy_struct_fields(&field_idents, &field_attrs, &field_ty);
+    let build_impl = impl_build(parent_name, &proxy_name, &field_idents, &field_attrs);
+
+    // TODO: decide if attrs are needed here too
+    quote! {
+        /// Proxy configuration structure to be used as an intermediate
+        /// for config loading
+        #[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
+        // #(#attrs)*
+        #vis struct #proxy_name #generics {
+            #proxy_fields
+        }
+
+        #load_env_impl
+        #disk_impl
+        #build_impl
+    }
+    .into()
+}
+
+fn impl_load_env(
+    struct_name: &Ident,
```

Comment:
Strongly recommend extracting the arguments into a single struct and using structured binding.

Reply:
Altered already to make `clippy` happier. Not too structured, but should be much more manageable with 4 instead of 8.